drivers/hmis/spec/models/hmis/ce/match/expression/cde_field_map_spec.rb

```diff
@@ -0,0 +1,132 @@
+
+  # Create a form definition and custom data element definition
+  let!(:form_definition) { create(:hmis_form_definition, identifier: 'test_form') }
+  let!(:string_cded) do
+    create(:hmis_custom_data_element_definition,
+           owner_type: 'Hmis::Hud::CustomAssessment',
+           key: 'language_preference',
+           field_type: 'string',
+           form_definition_identifier: 'test_form')
+  end
+  let!(:repeating_cded) do
+    create(:hmis_custom_data_element_definition,
+           owner_type: 'Hmis::Hud::CustomAssessment',
+           key: 'allergies',
+           field_type: 'string',
+           repeats: true,
+           form_definition_identifier: 'test_form')
+  end
+
+  describe '#client_query' do
+    before do
+      # Setup assessment for client1 - this one is older and should be ignored
+      old_assessment = create(:hmis_custom_assessment,
+                              client: client1,
+                              data_source: client1.data_source,
+                              assessment_date: current_date - 2.weeks,
+                              date_updated: current_date - 2.weeks,
+                              definition: form_definition)
+      create(:hmis_custom_data_element,
+             owner: old_assessment,
+             data_element_definition: string_cded,
+             value_string: 'Spanish',
+             data_source: client1.data_source)
+
+      # Setup the most recent assessment for client1
+      recent_assessment_c1 = create(:hmis_custom_assessment,
+                                    client: client1,
+                                    data_source: client1.data_source,
+                                    assessment_date: current_date - 1.week,
+                                    date_updated: current_date - 1.week,
+                                    definition: form_definition)
+      create(:hmis_custom_data_element,
+             owner: recent_assessment_c1,
+             data_element_definition: string_cded,
+             value_string: 'English',
+             data_source: client1.data_source)
+      create(:hmis_custom_data_element,
+             owner: recent_assessment_c1,
+             data_element_definition: repeating_cded,
+             value_string: 'Peanuts',
+             data_source: client1.data_source)
+      create(:hmis_custom_data_element,
+             owner: recent_assessment_c1,
+             data_element_definition: repeating_cded,
+             value_string: 'Dust',
+             data_source: client1.data_source)
+
+      # Setup assessment for client2
+      assessment_c2 = create(:hmis_custom_assessment,
+                             client: client2,
+                             data_source: client2.data_source,
+                             assessment_date: current_date - 1.week,
+                             date_updated: current_date - 1.week,
```

Comment:
Might be worth adding a data set where the date_updated is new but the assessment_date is old, to confirm it picks based on assessment date.

Reply:
* I'll adjust the spec to give confidence it's returning the most recent assessment
* we can take out date_updated here if we're sorting on assessment_date I think since it should not be relevant
* we can check the behavior when two assessments have the same assessment date that id is correctly used as a tie breaker